pkg/querier/http.go

```diff
@@ -199,7 +199,7 @@ func (q *QuerierAPI) SeriesHandler(ctx context.Context, req *logproto.SeriesRequ
 	logql.RecordSeriesQueryMetrics(ctx, utillog.Logger, req.Start, req.End, req.Groups, strconv.Itoa(status), req.GetShards(), statResult)
 
```

Comment:
Not going to block the PR on this, but I think we should return early here with nil check instead of doing the nil check at multiple places. Even better if we know resp would be nil only in case of err, and put an error check above at line 186 or 187.

Reply:
we might not be able to return early since the stats reporting happens for failed requests too. I moved this up though to avoid checks at multiple places